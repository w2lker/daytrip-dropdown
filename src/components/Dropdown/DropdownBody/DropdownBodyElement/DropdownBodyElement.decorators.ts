import withStyles from "react-jss";

import DropdownBodyElement from './DropdownBodyElement';
import dropdownBodyElementStyles from './DropdownBodyElement.styles';

const styledComponent = withStyles(dropdownBodyElementStyles)(DropdownBodyElement);

export default styledComponent;
