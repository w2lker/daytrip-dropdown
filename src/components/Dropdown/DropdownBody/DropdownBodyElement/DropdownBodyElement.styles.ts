const dropdownBodyElementStyles = {
  root: {
    fontWeight: 400,
    fontSize: 14,
    color: '#546F7B',
    paddingBottom: 13,
  },
  selected: {
    padding: 0,
    color: 'red',
  },
  focused: {
    padding: 0,
    color: 'blue',
  }
};

export default dropdownBodyElementStyles;
