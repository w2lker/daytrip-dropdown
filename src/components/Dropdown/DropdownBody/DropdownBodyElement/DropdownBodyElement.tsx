import React from 'react';

import classNames from "classnames";
import {WithStyles} from "react-jss";

import dropdownBodyElementStyles from "./DropdownBodyElement.styles";


export interface IDropdownBodyElementProps extends WithStyles<typeof dropdownBodyElementStyles>{
  content: string;
  selected?: boolean;
  focused?: boolean;
  onClick: () => void;
}
const emptyClasses = {
  root: '',
  selected: '',
  focused: '',
};

const DropdownBodyElement: React.FC<IDropdownBodyElementProps> = (props) => {
  const { content, selected, focused, onClick } = props;
  const classes = props.classes || emptyClasses;
  const wrapperClass = classNames({
    [classes.root]: true,
    [classes.focused]: focused,
    [classes.selected]: selected,
  });
  return (
    <div
      className={wrapperClass}
      onMouseDown={onClick}
    >
      {content}
    </div>
  );
};

export default DropdownBodyElement;
