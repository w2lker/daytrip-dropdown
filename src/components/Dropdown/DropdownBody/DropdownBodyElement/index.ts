import DropdownBodyElement from './DropdownBodyElement.decorators';

export default DropdownBodyElement;
