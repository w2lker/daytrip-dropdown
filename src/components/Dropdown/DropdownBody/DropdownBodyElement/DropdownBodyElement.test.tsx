import React from 'react';

import { mount, shallow } from 'enzyme';

import DropdownBodyElement, { IDropdownBodyElementProps } from './DropdownBodyElement';
import DropdownBodyElementDecorated from './DropdownBodyElement.decorators';
import dropdownBodyElementStyles from './DropdownBodyElement.styles';

const setup = () => {
  const classes = {
    root: 'dropdownBodyElement-wrapper',
    selected: 'dropdownBodyElement-selected',
    focused: 'dropdownBodyElement-focused',
  };
  const classesKeys = Object.keys(classes);
  const emptyProps: IDropdownBodyElementProps = {
    content: '',
    onClick: jest.fn(),
    classes,
  };
  const sampleProps: IDropdownBodyElementProps = {
    ...emptyProps,
    content: 'Some text',
    focused: true,
  };
  return { emptyProps, sampleProps, classes, classesKeys };
};

describe('DropdownBodyElement component', () => {

  it('match snapshot', () => {
    const { sampleProps } = setup();
    const component = shallow(<DropdownBodyElement {...sampleProps} />);
    expect(component.debug()).toMatchSnapshot();
  });

  it('displays content', () => {
    const { classes, sampleProps } = setup();
    const component = shallow(<DropdownBodyElement {...sampleProps}/>);
    expect(component.find(`.${classes.root}`).text()).toBe(sampleProps.content);
  });

  it('handles click', () => {
      const { classes, sampleProps } = setup();
      const component = shallow(<DropdownBodyElement {...sampleProps}/>);
      component.find(`.${classes.root}`).simulate('click');
      expect(sampleProps.onClick).toBeCalledTimes(1);
  });
});

describe('DropdownBodyElement styles', () => {
  it('contains principle classes', () => {
    const { classesKeys } = setup();
    expect(dropdownBodyElementStyles).toBeDefined();
    classesKeys.forEach((keyValue) => {
      // @ts-ignore
      expect(dropdownBodyElementStyles[keyValue]).toBeDefined();
    });
  });
});

describe('DropdownBodyElement decorators', () => {
  it('provides styled classes from decorators', () => {
    const { sampleProps, classesKeys } = setup();
    const component = mount(<DropdownBodyElementDecorated {...sampleProps} />);
    // @ts-ignore
    const assignedClasses = component.find('DropdownBodyElement').props().classes;
    classesKeys.forEach(keyValue => {
      expect(assignedClasses[keyValue]).toBeDefined();
    });
  });
});
