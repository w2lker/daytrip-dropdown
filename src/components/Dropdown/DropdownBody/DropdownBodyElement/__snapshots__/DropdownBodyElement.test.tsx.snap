// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownBodyElement component match snapshot 1`] = `
"<div className=\\"dropdownBodyElement-wrapper dropdownBodyElement-focused\\" onClick={[Function: mockConstructor]}>
  Some text
</div>"
`;
